pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -259,6 +260,26 @@ func doUpdateVol(ctx *volumemgrContext, status *types.VolumeStatus) (bool, bool)
 			changed = true
 			// Work is done
 			DeleteWorkCreate(ctx, status)
+			if status.MaxVolSize == 0 {
+				log.Infof("doUpdateVol: MaxVolSize is 0 for %s. Filling it up.",
+					status.FileLocation)
+				if status.IsContainer() {
+					size, err := utils.DirSize(status.FileLocation)
+					if err != nil {
+						log.Errorf("doUpdateVol: Computing size of %s failed: %v",
+							status.FileLocation, err)
+					} else {
+						status.MaxVolSize = uint64(size)
+					}
+				} else {
+					info, err := os.Stat(status.FileLocation)
+					if err != nil {
+						log.Errorf("doUpdateVol: Computing size of %s failed: %v",
+							status.FileLocation, err)
+					}
+					status.MaxVolSize = uint64(info.Size())
```

Comment:
Here isn't the max what we have as max in the qcow2 i.e., the same code as you just put in zedagent?

Reply:
Moved volume size logic in volume utils.